### DISCUSSION TITLE
Audio Downmixing, audio offset and disable subtitles

### DISCUSSION THREAD

**Discussion by ljransom** [moneytoo/Player#268] [Ideas]
Hello , a few things that would make this the perfect player.

1. Is it possible to include audio downmixing in this player? For example to downmix 5.1 to 2 channels for use on TV speakers. This is a great player which I would like to use as my goto player on all TVs, but with the lack of downmixing it's a bit of an issue for movies  when not using a surround sound system.

2. Adjust audio offset. Sometimes audio can be slightly out of sync, or go out of sync over time. This is an essential feature in my opinion so you can fix the audio sync yourself rather than having to get another file.

3. Option to disable subtitles by default. Currently you have to disable them every time you watch a video. It would be nice to have the player remember a default setting.

Thanks for all the work so far on this player.

**Comment by moneytoo** [comment]
1. As explained by the ExoPlayer devs in https://github.com/google/ExoPlayer/issues/7666, audio is downmixed if necessary. Alternatively, you can switch Just Player decoder priority to [device decoders](https://github.com/moneytoo/Player#:~:text=Prefer%20device%20or-,app%20decoders,-.) to force the use of FFmpeg.
2. This is not supported by ExoPlayer: https://github.com/google/ExoPlayer/issues/9947
3. Currently it follows the track preferences (default/forced) in video files. I already received few requests like this so I will think about possible improvements.

**Reply by ljransom** [reply]
Thanks for the response, shame about the offset not being supported. Audio definitely doesn't seem downmixed from what I've tested, but maybe that's just another limitation of exoplayer.

**Reply by moneytoo** [reply]
Trying out some channel test videos, audio downmixing works for me with both device and FFmpeg decoders. So this is probably some audio capability detection issue of ExoPlayer in case of your device.
